From 7a1e51f5d251015025c07afe0c6e711f644878b9 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Dec 2024 21:00:19 +0800
Subject: [PATCH 171/274] AOSCOS: mips: loongson64_cpuautoplug: adapt to
 bus_get_dev_root()

The original driver code uses the dev_root pointer in the bus_type struct.
However, with commit 9cc61e5fbd61 ("driver core: bus: move dev_root out of
struct bus_type"), the pointer was removed from the struct, encouraging
the use of bus_get_dev_root() instead.

Adapt to this change to fix build.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/loongson64/loongson64_cpuautoplug.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/arch/mips/loongson64/loongson64_cpuautoplug.c b/arch/mips/loongson64/loongson64_cpuautoplug.c
index eb129404379a..ad6220cb07b7 100644
--- a/arch/mips/loongson64/loongson64_cpuautoplug.c
+++ b/arch/mips/loongson64/loongson64_cpuautoplug.c
@@ -375,8 +375,11 @@ static struct platform_driver platform_driver = {
 static int __init cpuautoplug_init(void)
 {
 	int ret, delay;
+	struct device *dev_root;
 
-	ret = sysfs_create_group(&cpu_subsys.dev_root->kobj, &cpuclass_attr_group);
+	dev_root = bus_get_dev_root(&cpu_subsys);
+
+	ret = sysfs_create_group(&dev_root->kobj, &cpuclass_attr_group);
 	if (ret)
 		return ret;
 
@@ -410,9 +413,13 @@ static int __init cpuautoplug_init(void)
 
 static void __exit cpuautoplug_exit(void)
 {
+	struct device *dev_root;
+
+	dev_root = bus_get_dev_root(&cpu_subsys);
+
 	cancel_delayed_work_sync(&ap_info.work);
 	platform_driver_unregister(&platform_driver);
-	sysfs_remove_group(&cpu_subsys.dev_root->kobj, &cpuclass_attr_group);
+	sysfs_remove_group(&dev_root->kobj, &cpuclass_attr_group);
 }
 
 late_initcall(cpuautoplug_init);
-- 
2.47.1

